game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1587,16 +1588,19 @@ private void addCheckEndBattleAndRetreatingSteps(final List<IExecutable> steps)
     final BattleStep defensiveSubsRetreat = new DefensiveSubsRetreat(this, this);
     final BattleStep removeGeneralSuicide = new RemoveGeneralSuicide(this, this);
     final BattleStep offensiveGeneralRetreat = new OffensiveGeneralRetreat(this, this);
+    final BattleStep clearGeneralCasualties = new ClearGeneralCasualties(this, this);
 
-    steps.add(
-        new IExecutable() {
-          private static final long serialVersionUID = 8611067962952500496L;
+    steps.add(clearGeneralCasualties);
+    new IExecutable() {
+      private static final long serialVersionUID = 8611067962952500496L;
 
-          @Override
-          public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-            clearWaitingToDieAndDamagedChangesInto(bridge);
-          }
-        });
+      @Override
+      @RemoveOnNextMajorRelease
```

Comment:
We're going to be inline this on next major vs outright removal, right?
If so, this might help us in the future to comment that so we don't think we can *just* delete it:
```
RemoveOnNextMajorRelease("inline IExecutable body and remove IExecutable")
```

Reply:
No, it should be removed in the next major version.  These IExecutables are completely replaced by the `BattleStep` classes that I'm building.

Reply:
:+1: